aten/src/ATen/TensorIterator.cpp

```diff
@@ -760,7 +760,7 @@ int TensorIteratorBase::num_reduce_dims() const {
   return count;
 }
 
-void TensorIteratorBase::for_each(loop2d_t loop, int64_t grain_size) {
+void TensorIteratorBase::for_each(loop2d_t loop, int64_t grain_size) const {
```

Comment:
For foreach really now modify any internal TensorIterator states?

Reply:
The internal state is not modified.